packages/engine-test-utils/src/__tests__/engine-server/EngineEvents.spec.ts

```diff
@@ -0,0 +1,404 @@
+import { NoteChangeEntry, NoteChangeUpdateEntry } from "@dendronhq/common-all";
+import { NoteTestUtilsV4 } from "@dendronhq/common-test-utils";
+import { DendronEngineClient } from "@dendronhq/engine-server";
+import { createEngineFromServer, ENGINE_HOOKS, runEngineTestV5 } from "../..";
+
+/**
+ * Tests that the EngineEvents interface of the client-side DendronEngineClient signals properly
+ */
+describe("GIVEN a DendronEngineClient running on client-side", () => {
+  describe("WHEN writing a new note", () => {
+    test("THEN create event fired with the correct props AND update note event fired for its parent", async (done) => {
+      await runEngineTestV5(
+        async ({ engine, vaults, wsRoot }) => {
+          // TODO: Get rid of this casting once the proper way to expose EngineEvents of DendronEngineClient is implemented
+          const engineClient = engine as DendronEngineClient;
+
+          const newNote = await NoteTestUtilsV4.createNote({
+            fname: "alpha",
+            body: `[[beta]]`,
+            vault: vaults[0],
+            wsRoot,
+          });
+
+          engineClient.onEngineNoteStateChanged(
+            (noteChangeEntries: NoteChangeEntry[]) => {
+              try {
```

Comment:
why do we have try/catch here?

Reply:
Because we're in a callback, I found that if an `expect()` fails, the test will just hang (as it results in a failed promise instead of explicit thrown exception).  It's necessary to call `done(err)` for the test to properly fail when there's an expect error.

Reply:
got it. but for this specific test, are we expecting a failure? and if this is just a safety catch all, can we extract this in a common utility to reduce boilerplate code?

Reply:
We're not expecting a failure in the sense that the test case is expected to fail today, but we should expect a failure to potentially occur in the future if we have a product code bug.

imo - there isn't much to extract here (it's basically just a try catch block), but I pulled it out into a method with a header to describe why it's necessary.